platform/ios/src/MGLMapView.mm

```diff
@@ -1337,6 +1337,12 @@ - (void)handlePinchGesture:(UIPinchGestureRecognizer *)pinch
 
         [self unrotateIfNeededForGesture];
     }
+    
+    if ([self.delegate respondsToSelector:@selector(mapView:shouldChangeFromCamera:toCamera:)]
+        && ![self.delegate mapView:self shouldChangeFromCamera:oldCamera toCamera:self.camera])
+    {
+        self.camera = oldCamera;
+    }
```

Comment:
This is consistent with all the gesture handlers I implemented on macOS, as well as `-handlePanGesture:` above, but you’re right that it’s inconsistent with `-handleRotateGesture:` below.